src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+function checkOnError(...args: number[]): void {
+  if (args.some((arg: number) => arg <= 0)) {
+    throw new Error('Sides must be greater than 0');
+  }
+}
```

Comment:
Let's make the error message more general and then you can customize it more specifically for each shape type.

Example:
```
function checkSides(errorMessage: string, ...sides: number[]): void {
  if (sides.some(side => side <= 0)) {
    throw new Error(errorMessage);
  }
}
```